memgpt/functions/schema_generator.py

```diff
@@ -121,6 +122,10 @@ def generate_schema(function):
         if param.default == inspect.Parameter.empty:
             schema["parameters"]["required"].append(param.name)
 
+        if get_origin(param.annotation) is list:
+            if get_args(param.annotation)[0] is str:
+                schema["parameters"]["properties"][param.name]["items"] = {"type": "string"}
+
```

Comment:
@palandovalex is there a reason you didn't copy over the following from @Daenks 's PR?

https://github.com/cpacker/MemGPT/pull/661/files#diff-7dc915b0f453c308b7f29145e301571e37ae462b755ebf3cacdfb3d6db7ce903R99

```python
        if param.annotation == inspect.Parameter.empty:
            schema["parameters"]["required"].append(param.name)
```